### ISSUE TITLE
Ellis can keep hold of a MySQLdb lock

### ISSUE THREAD

**Issue by rkd-msw** [Metaswitch/ellis#141]
At https://github.com/Metaswitch/ellis/blob/dev/src/metaswitch/ellis/api/numbers.py#L126, if we throw an exception in the 'try' block, we'll never call `db_sess.commit()`. This means we keep the database locked, causing subsequent `sqlalchemy.exc.OperationalError: (_mysql_exceptions.OperationalError) (1205, 'Lock wait timeout exceeded; try restarting transaction')` errors.

Fix is either to use "autocommit()", or to structure the code like:

```
        try:
            number_id = numbers.allocate_number(db_sess, user_id, pstn)
            sip_uri = numbers.get_number(db_sess, number_id, user_id)
            ...
        except NotFound:
            # FIXME email operator to tell them we're out of numbers!
            _log.warning("No available numbers")
            db_sess.rollback()
            raise HTTPError(httplib.SERVICE_UNAVAILABLE,
                            "No available numbers")
        else:
            db_sess.commit()
```

or

```
        try:
            number_id = numbers.allocate_number(db_sess, user_id, pstn)
            sip_uri = numbers.get_number(db_sess, number_id, user_id)
            ...
        except NotFound:
            # FIXME email operator to tell them we're out of numbers!
            _log.warning("No available numbers")
            raise HTTPError(httplib.SERVICE_UNAVAILABLE,
                            "No available numbers")
        finally:
            db_sess.commit()
```

Easy to repro - truncate "ellis.numbers" in MySQL, try and create a new user (which fails), then immediately run `sudo /usr/share/clearwater/ellis/env/bin/python /usr/share/clearwater/ellis/src/metaswitch/ellis/tools/create_numbers.py --start 6505550000 --count 15`.

**Comment by rkd-msw** [comment]
Tagging @eleanor-merry .

Workaround is just to restart Ellis when you get into this state.

**Comment by rkd-msw** [comment]
Fixed by #142.
